### PR TITLE
Change default Views v2 AJAX method to POST

### PR DESCRIPTION
**Pull request by lucatume**
Issue: https://moderntribe.atlassian.net/browse/TEC-3283

[Screencast](https://drive.google.com/open?id=1GQEL67GVg1YltoJyEuMRQ1G30jBfvasB&authuser=luca%40tri.be&usp=drive_fs)

This PR changes the default method used by Views v2 to retrieve information from the backend from `GET` to `POST`.

This will avoid a number of issues with too long URLs (error 414) during AJAX requests without any behavioral change to the Views themselves.

### CODE REVIEW

File: src/Tribe/Views/V2/Rest_Endpoint.php [hunk 1]
```diff
@@ -188,13 +209,14 @@ public function enable_ajax_fallback() {
...
 	 * @since  4.9.7
+	 * @since TBD Changed the mock request HTTP method to POST (was GET).
 	 *
 	 * @param  array $params Associative array with the params that will be used on this mocked request
 	 *
 	 * @return Request The mocked request.
 	 */
 	public function get_mocked_rest_request( array $params ) {
-		$request = new Request( 'GET', static::ROOT_NAMESPACE . '/html' );
+		$request = new Request( 'POST', static::ROOT_NAMESPACE . '/html' );
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Rest_Endpoint.php:219]: :warning: Equals sign not aligned with surrounding assignments; expected 3 spaces but found 1 space

<sub>hash: 1782e1487691599d5deaab835a6ec696</sub>

File: src/Tribe/Views/V2/Rest_Endpoint.php [hunk 2]
```diff
@@ -224,11 +246,12 @@ public function get_mocked_rest_request( array $params ) {
...
 	 *
 	 * @since  4.9.7
-	 *
-	 * @return void
+	 * @since TBD Look up the POST data before the GET one to process the request.
 	 */
 	public function handle_ajax_request() {
-		$request = $this->get_mocked_rest_request( $_GET );
+		// Use the POST method data, if set; else fallback on the GET data.
+		$source = isset( $_POST ) ? $_POST : $_GET;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Rest_Endpoint.php:253]: * :warning: Equals sign not aligned with surrounding assignments; expected 2 spaces but found 1 space
* :no_entry_sign: Processing form data without nonce verification.
* :no_entry_sign: Processing form data without nonce verification.
* :warning: Processing form data without nonce verification.

<sub>hash: de57cc50f8ca156487c6240e30c93ec6</sub>

File: src/Tribe/Views/V2/View.php [hunk 3]
```diff
@@ -1405,6 +1406,9 @@ static function () {
 			}
 		);
 
+		/** @var Rest_Endpoint $endpoint */
+		$endpoint      = tribe( Rest_Endpoint::class );
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php:1410]: :warning: Equals sign not aligned correctly; expected 1 space but found 6 spaces

<sub>hash: d0b95eb1e4d2fc897af4254df8f1eaf6</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 2 error(s)
:warning: 4 warning(s)

File: src/Tribe/Views/V2/Rest_Endpoint.php [hunk 4]
```diff
@@ -224,11 +246,12 @@ public function get_mocked_rest_request( array $params ) {
...
 	 *
 	 * @since  4.9.7
-	 *
-	 * @return void
+	 * @since TBD Look up the POST data before the GET one to process the request.
 	 */
 	public function handle_ajax_request() {
-		$request = $this->get_mocked_rest_request( $_GET );
+		// Use the POST method data, if set; else fallback on the GET data.
+		$source  = isset( $_POST ) ? $_POST : $_GET;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Rest_Endpoint.php:253]: * :no_entry_sign: Processing form data without nonce verification.
* :no_entry_sign: Processing form data without nonce verification.
* :warning: Processing form data without nonce verification.

<sub>hash: 61904cd92e27a45c75e15386f8e25046</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 2 error(s)
:warning: 1 warning(s)

**Review by paulmskim — DISMISSED** [review]
FE looks good, BE looks good from what I can see.

**Review by bordoni — DISMISSED** [review]
Do we have a PR with modifications to the PRO views? Related to: `data-view-rest-method`

**Comment by lucatume** [comment]
> Do we have a PR with modifications to the PRO views? Related to: `data-view-rest-method`

https://github.com/moderntribe/events-pro/pull/1595

File: src/Tribe/Views/V2/Rest_Endpoint.php [hunk 5]
```diff
@@ -300,4 +323,27 @@ public static function did_rest_authentication_errors( $errors = null ) {
 
 		return $errors;
 	}
+
+	/**
+	 * Returns the filtered HTTP method Views should use to fetch their content from the backend endpoint..
```
Comment (Camwyn) [inline src/Tribe/Views/V2/Rest_Endpoint.php:328]:
```suggestion
	 * Returns the filtered HTTP method Views should use to fetch their content from the backend endpoint.
```

File: src/Tribe/Views/V2/Rest_Endpoint.php [hunk 6]
```diff
@@ -300,4 +323,27 @@ public static function did_rest_authentication_errors( $errors = null ) {
...
+	 * @return string The filtered HTTP method Views should use to fetch their content from the back-end endpoint.
+	 */
+	public function get_method() {
+		/**
+		 * Filters the HTTP method Views should use to fetch their contents calling the back-end endpoint.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $method The HTTP method Views will use to fetch their content. Either `POST` (default) or
+		 *                       `GET`. Invalid values will be defaulted to `POST`.
```
Comment (Camwyn) [inline src/Tribe/Views/V2/Rest_Endpoint.php:341]:
```suggestion
		 *                       `GET`. Invalid values will be set to the default `POST`.
```

**Review by Camwyn — CHANGES_REQUESTED** [review]
Couple small docblock tweaks and a question:
We cover GET for BC - should we have a test to cover that BC?

**Review by jesseeproductions — APPROVED** [review]

**Review by bordoni — DISMISSED** [review]

**Pull request merged**